packages/app/src/files-manager/actions/upload-file.action.ts

```diff
@@ -45,9 +47,33 @@ export const uploadFile = createAsyncThunk(
       return rejectWithValue(filesToUpload.error)
     }
 
-    const filePaths = filesToUpload.data
+    const filePaths = filesToUpload.data ?? []
 
-    if (filePaths?.length === 0) {
+    if (filePaths.length === 0) {
+      dispatch(setUploadBlocked(false))
+      return
+    }
+
+    const harmonyFreeFilesSlotsCount = getFreeFilesSlotsCountForHarmony(
```

Comment:
I suggest creating variables and functions similar to keep a project easy to read. My algorithm:

```
get + harmonyFreeFilesSlotsCount = verb + noun
```

```suggestion
    const harmonyFreeFilesSlotsCount = getHarmonyFreeFilesSlotsCount(
```

Reply:
Resolved